e2etest/slasher_e2e_test.go

```diff
@@ -283,8 +213,14 @@ func TestSlasher_SlashingUnbonding(t *testing.T) {
 		_, err := tm.BTCClient.GetRawTransaction(unbondingSlashingMsgTxHash)
 		return err == nil
 	}, eventuallyWaitTimeOut, eventuallyPollTime)
+
 	// mine a block that includes slashing tx
-	tm.MineBlockWithTxs(t, tm.RetrieveTransactionFromMempool(t, []*chainhash.Hash{unbondingSlashingMsgTxHash}))
+	require.Eventually(t, func() bool {
```

Comment:
Would be nice to wrap up these eventually check into a function inside the test manager, but is only a suggestion